functionary/ui/templates/partials/environments/environment_management.html

```diff
@@ -0,0 +1,54 @@
+<label class="label" for="users">
+    <i class="fa fa-user"></i>&nbsp;Users:
+    <button class="button is-small has-text-link is-white singletonActive"
+            hx-trigger="click"
+            hx-get="{% url 'ui:create-environment-member' environment_id %}"
+            hx-target="#add-user"
+            type="button">
+        <span class="fa fa-plus"></span>
+    </button>
+</label>
+
+<div id="add-user">
+</div>
+
+<table class="table is-hoverable">
+    <thead>
+        <tr>
+            <th>Name</th>
+            <th>Username</th>
+            <th>Effective Role</th>
+            <th>Role Origin</th>
+            <th></th>
+            <th></th>
+        </tr>
+    </thead>
+    <tbody>
+        {% for u in users %}
+            <tr>
+                <td>{{ u.user.first_name }} {{ u.user.last_name }}</td>
+                <td>{{ u.user.username }}</td>
+                <td>{{ u.role }}</td>
+                <td>{{ u.origin }}</td>
+                <td>
+                    <button class="button is-small has-text-link is-white singletonActive"
+                            hx-get="{% url 'ui:update-environment-member' environment_id=environment_id user_id=u.user.id %}"
+                            hx-target="#update-modal">
+                        <span class="fa fa-wrench"></span>
+                    </button>
+                </td>
+                <td>
+                    <button class="button is-small has-text-link is-white singletonActive"
+                            hx-delete="{% url 'ui:delete-environment-member' environment_id=environment_id user_id=u.user.id %}"
```

Comment:
I think I will just go with the `HX-Refresh` header for the time being. If I wanted to avoid needing to re-render the whole page or the table, I could just make a partial for the row and return that partial depending if the user has actually been removed from the environment or not.